### PR TITLE
Refine error message for thread-safe usage of std::sync::Arc<{Self}>

### LINKED ISSUE #114687: Suggestion to use Arc<T> is too indiscriminate
### Code

```rust
fn main() {
    let ptr = &false as *const bool;

    std::thread::spawn(move || println!("{:p}", ptr)).join().unwrap();
}
```

### Current output

```console
error[E0277]: `*const bool` cannot be sent between threads safely
 --> src/main.rs:4:24
  |
4 |     std::thread::spawn(move || println!("{:p}", ptr)).join().unwrap();
  |     ------------------ -------^^^^^^^^^^^^^^^^^^^^^^
  |     |                  |
  |     |                  `*const bool` cannot be sent between threads safely
  |     |                  within this `[closure@src/main.rs:4:24: 4:31]`
  |     required by a bound introduced by this call
  |
  = help: within `[closure@src/main.rs:4:24: 4:31]`, the trait `Send` is not implemented for `*const bool`
  = note: consider using `std::sync::Arc<*const bool>`; for more information visit <https://doc.rust-lang.org/book/ch16-03-shared-state.html>
note: required because it's used within this closure
 --> src/main.rs:4:24
  |
4 |     std::thread::spawn(move || println!("{:p}", ptr)).join().unwrap();
  |                        ^^^^^^^
note: required by a bound in `spawn`
 --> /rustc/08d00b40aef2017fe6dba3ff7d6476efa0c10888/library/std/src/thread/mod.rs:680:1
```

### Desired output

_No response_

### Rationale and extra context

Filing to track the hesitation from https://github.com/rust-lang/rust/pull/88936#pullrequestreview-775249086.

> [@nagisa]&mdash; Use of `Arc` is not always applicable, as both `Send` and `Sync` impls for `Arc<T>` require `T: Send + Sync`. Unless we can verify `T` bounds here, I don't think this `on` clause adds any more clarity over the default `note`?

> [@estebank]&mdash; Thats fair, but was concerned the default would be too verbose for a case where we could be more straightforward. I can change it.

> [@nagisa]&mdash; Well, we can always change this later too. r=me if you don't want to change it.

If T is not Send then **Arc\<T\> is also not Send**. Suggesting Arc\<T\> whenever Send is not implemented seems misguided.

### Other cases

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by reez12g**
fixes https://github.com/rust-lang/rust/issues/114687

### CODE REVIEW

**Comment by estebank** [comment]
We can leave only the link to the book and not mention Arc at all in the !Send case.

**Comment by thomcc** [comment]
I don't have strong feelings here, so perhaps @m-ou-se should take this review, since she seems to care more than I.

r? @m-ou-se

File: library/core/src/marker.rs [hunk 1]
```diff
@@ -79,7 +79,7 @@ macro marker_impls {
     on(_Self = "std::rc::Rc<T, A>", note = "use `std::sync::Arc` instead of `std::rc::Rc`"),
     message = "`{Self}` cannot be sent between threads safely",
     label = "`{Self}` cannot be sent between threads safely",
-    note = "consider using `std::sync::Arc<{Self}>`; for more information visit \
+    note = "consider whether `std::sync::Arc<{Self}>` could be incorporated to share this value between threads; for more information visit \
```
Comment (dtolnay) [inline library/core/src/marker.rs:82]: As long as `Arc<{Self}>` continues to be mentioned, this does not fix #114687. That suggestion is *always* wrong.

**Review by dtolnay — CHANGES_REQUESTED** [review]

**Comment by reez12g** [comment]
Thank you for your reviews. 
Since issue #114687 is now closed, I am going to close this Pull Request as well.

**Pull request closed without merging**